assignments/Assignment 11.txt

```diff
@@ -0,0 +1,49 @@
+Assignment 11
+
+
+1) Rewrite the following function into Lambda syntax:
+function largestOfThree(a, b, c) {
+if (a > b && a > c) return a;
+if (b > a && b > c) return b;
+return c;
+}
+
+const largestOfThree = (a, b, c) ==> {
+if (a > b && a > c) {return a};
+if (b > a && b > c) {return b};
+{return c};
+}
+
+
+2) Rewrite your previous program that called an API so that it uses lambda syntax.
+See attached
+
+3) Explain in English how this recursive function runs:
+function sum(n) {
+if (n === 1) {
+return 1;
+}
+return n + sum(n - 1);
+}
+
+A number is input as a variable to function called sum as (n). For example, sum(5). The if statement defines the last byte so the function terminates and returns the final answer in return 1. Otherwise the function reiterates over each number. For example, return 5 (n) + sum(5-1)
+5 + 4
```

Comment:
Actually the sum would proceed from the lower numbers towards the higher numbers:

1
2 + 1
3 + 3
4 + 6
5 + 10

Try to see why this is.